### ISSUE TITLE
array display properties in points is just "bad"

### ISSUE THREAD

**Issue by porres** [plugdata-team/plugdata#1677]
In Vanilla, when we display with points we can see a sample quite nicely, but plugdata's point view is jut bad and can only show some points.

Taking as an example the [soundilfer] help file below

<img width="938" alt="Screenshot 2024-06-25 at 21 47 57" src="https://github.com/plugdata-team/plugdata/assets/5360142/59303e0b-9884-4042-b13e-78123f4da721">

**Comment by dromer** [comment]
I think the PD view of the table is misleading, since we can't actually see any points. So the "can only show some points" is the correct view for what you'd expect for "points". These black bars in the vanilla view are kind of weird. It may look "better", but how are these "points"?

The bezier and polygon views are pretty much identical and make more sense.

Imo this is a case where we should maybe request the vanilla view to show proper points instead of this odd black mass.

**Comment by porres** [comment]
No one in their right mind would consider changing the way Pd looks and relies on graphically and break things and simply ruin sample visualization for no good reason just because one might challenge the conceptual meaning of points... it'd make more sense to change the name perhaps to "weird and odd black mass"... but 'points' does make sense since for smaller table sizes you can see the points and they are in fact actual points. You just cannot see 'every point' when there are too many points, and the same is true for any digital sound editor, cause you can't see the sample points unless you zoom in drastically. 

<img width="275" alt="Screenshot 2024-06-26 at 02 50 49" src="https://github.com/plugdata-team/plugdata/assets/5360142/5891bb1b-0a83-4f91-9edf-0981f0d3d180">

**Comment by dromer** [comment]
The plugdata display is simply more precise, so you don't get these visual artifacts.

So you want the display to be _less_ precise and as ugly as vanilla? Did I mention stockholm syndrome yet or am I just not in the right mind?

**Comment by dromer** [comment]
Arguably both of these views are incorrect and instead there should be the classic "lolly-pop" view.

Something called "points" should always display points. Or should there be some cut-off size at which it suddenly becomes a black blob?

**Comment by dromer** [comment]
I would agree that it could deal with the specific table resolution better, that is misleading in the plugdata view in this specific instance (there are way more points that it looks like in the render).

**Comment by dromer** [comment]
Since these are both abstracted views of the underlying data neither of these are precise or "correct".

Plugdata is "consistent" in that it shows an abstracted set of "points", however the main inconsistency here is that you'd expect to see more points as you zoom in (to fit the pixels on the screen) which is not the case

**Comment by ben-wes** [comment]
> The plugdata display is simply more precise, so you don't get these visual artifacts.

hmm ... puredata is drawing all the points (or at least: drawing each pixel when there is a point), resulting in a shape that doesn't resemble points. might be called an artifact - but that result is expected in this case and gives a visualization of a filled envelope. plugdata is skipping points (not sure how it chooses the ones to visualize - since that doesn't change with zooming). but i wouldn't call skipping information more precise.

puredata arguably does the same skipping for lines/bezier ... certainly for performance reasons.

> [...] there should be the classic "lolly-pop" view.

that would be a nice addition for sure and faster to draw. is that derived from the max amplitude per visual step?

**Comment by ben-wes** [comment]
> In the case of vanilla they choose to show lines instead of points when there is more data than pixels (or so it seems).

i think it's the logic as described (color the pixel if there is a point inside). you can see the single points when enlarging the area of the array ... i wish i could just zoom in there. ;)

**Comment by ben-wes** [comment]
> i think it's the logic as described (color the pixel if there is a point inside)

i withdraw this statement ... it's not just points. it seems more like what you said: it draws lines when there are too many points (on the screenshot, there should only be a single point at the bottom otherwise). so there could be improvements an all sides ...
zooming in with plugdata would be great. not sure how to get these visualizations completely right ...

![image](https://github.com/plugdata-team/plugdata/assets/1923787/c4e03ba3-e4c1-4914-9698-360cd4c4fb62)

**Comment by dromer** [comment]
> zooming in with plugdata would be great

yeah I think the fact that we can zoom is an opportunity to improve on these kind of renderings with more detail

**Comment by ben-wes** [comment]
> both renders are too course to be really informative at this array resolution.

puredata is obviously using a little hack here to achieve the typical amplitude envelope view by drawing these vertical lines from min to max value that are inside each horizontal step.

it's certainly not a point view anymore then (and possibly shouldn't be called that way) - but it does transport a bit more information that way (min and max values instead of arbitrary single value per bin).

**Comment by porres** [comment]
> So you want the display to be less precise and as ugly as vanilla? Did I mention stockholm syndrome yet or am I just not in the right mind?

"Ugly", properly showing a sample now is "ugly", hahahahahahahahaha...

yeah, let's make plugdata totally incompatible and plot a completely useless and non functional sample plot because you think it's pretty.

Dude, you're right, we cannot get mad at you, cause you're actually joking, right? and I get it, it's hilarious.

Either that or you're completely out of your mind, which is funny too.

**Comment by porres** [comment]
> puredata is obviously using a little hack here to achieve the typical amplitude envelope view by drawing these vertical lines from min to max value that are inside each horizontal step.

no, it's not obvious, it's "ugly", the obvious thing is that you're suffering from stockholm syndrome... the best idea is just to completely ignore this functionality and offer a completely useless solution for plotting samples, not to mention ruin compatibility completely, which is also a great idea. People using Vanilla should use PlugData to learn how it's actually flawed and stop using it and throw away their patches and use a much better and more intelligente software, this fork must be a spiritual guide that frees people from stockholm syndrome and other issues. 

Fuck Miller, let's take his work, fool people into saying things should be compatible but just offer a much better and superior software that they don't know yet cause they're part of a cult or something, let's just attack Pd's community, divide it, and be dicks, that's clever and very productive.

**Comment by dromer** [comment]
@porres stop being such a fucking dick, you psychopath.

**Comment by porres** [comment]
it's not psychopathy, it's stockholm syndrome, ins't it? You're quite the mental health expert, full of diagnosis, please let me know exactly what I have, one, the other, both, maybe even more stuff.

Let me tell you what I think it's being a dick, you know... trying hard to go out of your way just so you can complain about vanilla and try to promote inconsistent and incompatible behaviour... not to mention the attitude.

I know this is not the route and mentality of this project, if it were, I wouldn't be here and it'd be the same successful route of Pd-L2ork/Purr Data by the way, and I guess you will find heaven if you collaborate with them instead :) note that they  have a lollypop and/or bar view or something for arrays.

Anyway, bottom line:
- point view in arrays is used in Pd to plot soundfiles, this won't change! If you find it misleading you can propose a better way to document it... I'm actually thinking about that right now, though I never considered necessary.
- PlugData should simply follow suit in this case, unless you wanna fuck around even worse and to an extent that even Pd-l2ork and Purr Data people never 'dreamed' or dared to...
- If you want a new and better array plotter with other fancy functionalities and a different behaviour you can design a new external and just leave the original alone. I can include that in ELSE for compatibility.

**Comment by porres** [comment]
<img width="594" alt="Screenshot 2024-06-26 at 09 55 55" src="https://github.com/plugdata-team/plugdata/assets/5360142/1c963b73-f62c-4a6f-b390-bf892857ea08">

**Comment by porres** [comment]
closing and rebooting this as it's gone sour and comments have been deleted... sorry to see it escalated when we only had innocent utter jokes, I will try to avoid having a light-hearted conversation next time.
